src/operator/tensor/matrix_op-inl.h

```diff
@@ -653,6 +653,7 @@ inline void GetIndexRange(const TShape& dshape,
   }
 
   for (index_t i = 0; i < param_begin.ndim(); ++i) {
+    CHECK_NE(dshape[i], 0) << "Invalid input shape!";
```

Comment:
0 means the dimension hasn't been successfully inferred yet which doesn't necessarily mean the shape is invalid.

Reply:
Agreed. Will consider it a valid input.